### DISCUSSION TITLE
How to use paginate data with more than one model? [Adonis 5]

### DISCUSSION THREAD

**Discussion by pk4all** [adonisjs/core#1756] [General]
Hi,
I want to search data with two models ``Post`` and ``User`` and want to paginate both result.

**Comment by melokki** [comment]
Hello,
Have you tried something like this?

```js
await User.query()
        .where("name", "LIKE", "%something%")
        .preload("posts", (query) => {
          query.where('title', "LIKE", "%something%");
        }).paginate(request.input('page', 1),10);
```

**Comment by thetutlage** [comment]
Why would you merge the data from Users and Posts into one object? If you are looking for relationships, then @melokki has answered it for you

**Reply by pk4all** [reply]
Yes I need Users and Posts into one object, In my app we perform a search operation from User and post both table. So we need this.

**Reply by thetutlage** [reply]
I suggest using the database query builder directly and perform the join. Merging results of two models into one is not possible

**Reply by pk4all** [reply]
> I suggest using the database query builder directly and perform the join. Merging results of two models into one is not possible

@thetutlage Okay Thanks a lot for quick reply.
